# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/compiler/translator/tree_ops/EmulatePrecision.cpp
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/compiler/translator/tree_ops/EmulatePrecision.cpp
--- a/gfx/angle/checkout/src/compiler/translator/tree_ops/EmulatePrecision.cpp	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/compiler/translator/tree_ops/EmulatePrecision.cpp	Sat Sep 07 13:38:36 2019 +0000
@@ -174,7 +174,7 @@
 
 std::string RoundingHelperWriterESSL::getTypeString(const char *glslType)
 {
-    std::stringstream typeStrStr;
+    std::stringstream typeStrStr = sh::InitializeStream<std::stringstream>();
     typeStrStr << "highp " << glslType;
     return typeStrStr.str();
 }
@@ -257,7 +257,7 @@
 void RoundingHelperWriterGLSL::writeVectorRoundingHelpers(TInfoSinkBase &sink,
                                                           const unsigned int size)
 {
-    std::stringstream vecTypeStrStr;
+    std::stringstream vecTypeStrStr = sh::InitializeStream<std::stringstream>();
     vecTypeStrStr << "vec" << size;
     std::string vecType = getTypeString(vecTypeStrStr.str().c_str());
 
@@ -287,7 +287,7 @@
                                                          const unsigned int rows,
                                                          const char *functionName)
 {
-    std::stringstream matTypeStrStr;
+    std::stringstream matTypeStrStr = sh::InitializeStream<std::stringstream>();
     matTypeStrStr << "mat" << columns;
     if (rows != columns)
     {
@@ -379,7 +379,7 @@
 void RoundingHelperWriterHLSL::writeVectorRoundingHelpers(TInfoSinkBase &sink,
                                                           const unsigned int size)
 {
-    std::stringstream vecTypeStrStr;
+    std::stringstream vecTypeStrStr = sh::InitializeStream<std::stringstream>();
     vecTypeStrStr << "float" << size;
     std::string vecType = vecTypeStrStr.str();
 
@@ -409,7 +409,7 @@
                                                          const unsigned int rows,
                                                          const char *functionName)
 {
-    std::stringstream matTypeStrStr;
+    std::stringstream matTypeStrStr = sh::InitializeStream<std::stringstream>();
     matTypeStrStr << "float" << columns << "x" << rows;
     std::string matType = matTypeStrStr.str();
 
@@ -744,7 +744,7 @@
                                                                              TIntermTyped *right,
                                                                              const char *opNameStr)
 {
-    std::stringstream strstr;
+    std::stringstream strstr = sh::InitializeStream<std::stringstream>();
     if (left->getPrecision() == EbpMedium)
         strstr << "angle_compound_" << opNameStr << "_frm";
     else